ctapipe/io/dl1writer.py

```diff
@@ -151,7 +151,14 @@ class DL1Writer(Component):
     peak_time_offset = Int(default_value=0).tag(config=True)
     peak_time_scale = Float(default_value=100.0).tag(config=True)
 
-    def __init__(self, event_source: EventSource, config=None, parent=None, **kwargs):
+    def __init__(
+        self,
+        event_source: EventSource,
+        config=None,
+        parent=None,
+        is_muon=False,
```

Comment:
I don't think you need this attribute. The DL1Writer should never care if this is a muon run or not. It should write muon data if there are muon data in the event, if if not, it skips. it.